experimental/sql/src/impl_to_iterators.py

```diff
@@ -50,24 +53,84 @@ class FullTableScanRewriter(RelImplRewriter):
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.FullTableScanOp,
                         rewriter: PatternRewriter):
-    # TODO: Change this once not only SampleInputOp is supported
+    # TODO: Change this once loading from input is supported
     rewriter.replace_matched_op(
-        it.SampleInputOp.get(self.convert_bag(op.result.typ)))
+        it.ConstantStreamOp.get([[IntegerAttr.from_int_and_width(0, 32)],
+                                 [IntegerAttr.from_int_and_width(1, 32)],
+                                 [IntegerAttr.from_int_and_width(2, 32)],
+                                 [IntegerAttr.from_int_and_width(3, 32)]],
+                                self.convert_bag(op.result.typ)))
 
 
 @dataclass
 class AggregateRewriter(RelImplRewriter):
 
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.Aggregate, rewriter: PatternRewriter):
-    rewriter.replace_matched_op(it.ReduceOp.get(op.input.op))
+    rewriter.replace_matched_op(
+        it.ReduceOp.get(op.input.op, StringAttr.from_str("sum_struct"),
+                        self.convert_bag(op.result.typ)))
 
 
 #===------------------------------------------------------------------------===#
 # Conversion setup
 #===------------------------------------------------------------------------===#
 
 
+def add_reduce_functions(ctx: MLContext, mod: ModuleOp):
+  sum_struct = FuncOp.from_region(
+      "sum_struct",
+      [
+          LLVMStructType.from_type_list([IntegerType.from_width(32)]),
+          LLVMStructType.from_type_list([IntegerType.from_width(32)])
+      ],
+      [LLVMStructType.from_type_list([IntegerType.from_width(32)])],
+      Region.from_block_list(
+          [
+              Block.from_callable(
+                  [
+                      LLVMStructType.from_type_list(
+                          [IntegerType.from_width(32)]),
+                      LLVMStructType.from_type_list(
+                          [IntegerType.from_width(32)])
+                  ],
+                  lambda ba1, ba2: [
+                      l := LLVMExtractValue.build(
+                          result_types=[IntegerType.from_width(32)],
+                          attributes={
+                              "position":
+                                  ArrayAttr.from_list(
+                                      [IntegerAttr.from_index_int_value(0)])
+                          },
```

Comment:
Maybe you can bind the integer type and the index attribute to a local variable and factor those expressions out to avoid repetition.